### PR TITLE
Clean-up http code.

### PR DESCRIPTION
**Pull request by croudet**

### CODE REVIEW

File: http-netty/src/main/java/io/micronaut/http/netty/reactive/HandlerPublisher.java [hunk 1]
```diff
@@ -259,13 +259,13 @@ public void channelRegistered(ChannelHandlerContext ctx) {
     private void provideChannelContext(ChannelHandlerContext ctx) {
         switch (state) {
             case NO_SUBSCRIBER_OR_CONTEXT:
-                verifyRegisteredWithRightExecutor(ctx);
+                verifyRegisteredWithRightExecutor();
```
Comment (jameskleeh) [inline http-netty/src/main/java/io/micronaut/http/netty/reactive/HandlerPublisher.java:262]: I don't think this change is correct. The instance variable isn't set until after the method is called

Comment (croudet) [reply http-netty/src/main/java/io/micronaut/http/netty/reactive/HandlerPublisher.java:262]: `ctx` was not used in `verifyRegisteredWithRightExecutor`

Comment (jameskleeh) [reply http-netty/src/main/java/io/micronaut/http/netty/reactive/HandlerPublisher.java:262]: Ah I see. Github collapsed some code and it looked like it was. Thanks

**Comment by jameskleeh** [comment]
Thanks!

**Pull request merged**